lib/timers.js

```diff
@@ -696,6 +705,7 @@ function createImmediate(args, callback) {
 
 
 exports.clearImmediate = function(immediate) {
+  // TODO: also verify `immediate instanceof Immediate`?
```

Comment:
the linkedlist remove should already be guarding against problems, seems unnecessary